src/main/java/no/unit/nva/cristin/projects/FetchCristinProjects.java

```diff
@@ -45,75 +36,70 @@
     private static final String CRISTIN_QUERY_PARAMETER_PER_PAGE_KEY = "per_page";
     private static final String CRISTIN_QUERY_PARAMETER_PER_PAGE_VALUE = "5";
 
-    private transient CristinApiClient cristinApiClient;
+    public static final String LANGUAGE_QUERY_PARAMETER = "language";
+    public static final String TITLE_QUERY_PARAMETER = "title";
+    private static final String DEFAULT_LANGUAGE_CODE = "nb";
+    private final transient CristinApiClient cristinApiClient;
     private final transient PresentationConverter presentationConverter = new PresentationConverter();
 
+    @SuppressWarnings("unused")
     public FetchCristinProjects() {
-        cristinApiClient = new CristinApiClient();
+        this(new CristinApiClient(), new Environment());
     }
 
-    public FetchCristinProjects(CristinApiClient cristinApiClient) {
+    public FetchCristinProjects(CristinApiClient cristinApiClient, Environment environment) {
+        super(Void.class, environment, LoggerFactory.getLogger(FetchCristinProjects.class));
         this.cristinApiClient = cristinApiClient;
     }
 
-    public void setCristinApiClient(CristinApiClient cristinApiClient) {
-        this.cristinApiClient = cristinApiClient;
+    @Override
+    protected ProjectPresentation[] processInput(Void input, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+
+        return createProjectPresentations(checkLanguageAndFetch(requestInfo), checkTitleAndFetch(requestInfo));
     }
 
     @Override
-    @SuppressWarnings("unchecked")
-    public GatewayResponse handleRequest(Map<String, Object> input, Context context) {
-
-        GatewayResponse gatewayResponse = new GatewayResponse();
-        try {
-            this.checkParameters(input);
-        } catch (RuntimeException e) {
-            gatewayResponse.setErrorBody(e.getMessage());
-            gatewayResponse.setStatusCode(Response.Status.BAD_REQUEST.getStatusCode());
-            return gatewayResponse;
-        }
-
-        Map<String, String> queryStringParameters = (Map<String, String>) input.get(QUERY_STRING_PARAMETERS_KEY);
-        String title = queryStringParameters.get(TITLE_KEY);
-        String language = queryStringParameters.getOrDefault(LANGUAGE_KEY, DEFAULT_LANGUAGE_CODE);
+    protected Integer getSuccessStatusCode(Void input, ProjectPresentation[] output) {
+        return HttpURLConnection.HTTP_OK;
+    }
 
-        try {
-            Map<String, String> cristinQueryParameters = createCristinQueryParameters(title, language);
+    private String checkTitleAndFetch(RequestInfo requestInfo) throws BadRequestException {
```

Comment:
getValidTitle (I don't like methods that tell me they do multiple things with "and" as this tends to mean the method is violating the SRP), I also tend to reserve "fetch" for HTTP-GETs.

Reply:
should i only change name or split logic into one validator and one get value as well? I wanted to do several things with less code, and calling this::isValid implicit made the code a little more compact